azure-storage-cpp-lite/include/blob/blob_client.h

```diff
@@ -414,7 +414,8 @@ namespace microsoft_azure { namespace storage {
         /// <param name="size">The size of the data to download from the blob, in bytes.</param>
         /// <param name="destPath">The target file path.</param>
         /// <param name="parallel">A size_t value indicates the maximum parallelism can be used in this request.</param>
-        void download_blob_to_file(const std::string &container, const std::string &blob, const std::string &destPath, size_t parallel = 9);
+        /// <returns>A <see cref="storage_outcome" /> object that represents the properties (etag, last modified time and size) from the first chunk retrieved.</returns>
+        storage_outcome<chunk_property> download_blob_to_file(const std::string &container, const std::string &blob, const std::string &destPath, size_t parallel = 9);
```

Comment:
common, why download_blob_to_file returns chunk_property? let's introduce new type?

Reply:
Because it is the chunk_property. We retrieve these properties from the first chunk. Should I use blob_property instead ?

Reply:
or create a new type containing size, etag and last modification time.

Reply:
There is 2 additional issues:
1. previously storage_output<> was not exposed to blobfuse
2. interface becoming inconsistent - it should either setup errno as it does now or should return strorage_output from all methods or should use exceptions.
Let's decide which approach to use and stay with it? In our current approach the download_file_... should return nothing, etag and last modification time should be output parameters, error should be returned with errno.
I will be happy if you change the interface to throw exceptions or return storage_outcome from each method